### ISSUE TITLE
Data Binding Error with Open Records with Numbers

### ISSUE THREAD

**Issue by lafernando** [ballerina-platform/ballerina-standard-library#332]
Consider the following sample service:

```ballerina
import ballerina/http;
import ballerina/io;

type Entry record {
    string src;
    string ref;
};

@http:ServiceConfig {
    basePath: "/"
}
service testSvc on new http:Listener(9090) {

    @http:ResourceConfig {
        path:"/store",
        methods: ["POST"],
        body: "entry"
    }
    resource function store(http:Caller caller, http:Request request, Entry entry) returns error? {
        io:println("Input: ", entry);
        check caller->ok();
    }

}
```

And the following requests:

```bash
$ curl -d '{"src":"s1","ref":"r1"}' http://localhost:9090/store
$ curl -d '{"x":1,"src":"s1","ref":"r1"}' http://localhost:9090/store
$ curl -d '{"x":1.3,"src":"s1","ref":"r1"}' http://localhost:9090/store
data binding failed: error("{ballerina/lang.typedesc}ConversionError",message="'map<json>' value cannot be converted to 'Entry'")
```

With the open record, if we set a numerical value to a field that is not mentioned there in the record, with a fractional value, it fails with the above error. 

Version: Ballerina slp4.

**Comment by chamil321** [comment]
This behaviour has induced due to https://github.com/ballerina-platform/ballerina-lang/issues/26624. 
A possible workaround would be to explicitly define the decimal field and make it optional in the `Entry` record as follows until the bug is fixed.
```
type Entry record {
    decimal x?;
    string src;
    string ref;
};
```

**Comment by chamil321** [comment]
Tested issue with 
```
Ballerina Swan Lake Beta 1
Language specification 2021R1
Update Tool 1.3.5
```

```ballerina
import ballerina/http;
import ballerina/io;

type Entry record {
    string src;
    string ref;
};

service on new http:Listener(9090) {
    resource function post store(@http:Payload Entry entry) returns http:Ok|error? {
        io:println("Input: ", entry);
        return {};
    }
}
```
Output
```
[ballerina/http] started HTTP/WS listener 0.0.0.0:9090
Input: {"src":"s1","ref":"r1"}
Input: {"src":"s1","ref":"r1","x":1}
Input: {"src":"s1","ref":"r1","x":1.3}
```

The issue has been resolved via ballerina-platform/ballerina-lang#26624., hence closing the issue
